globset/src/lib.rs

```diff
@@ -567,7 +567,7 @@ impl LiteralStrategy {
     }
 
     fn add(&mut self, global_index: usize, lit: String) {
-        self.0.entry(lit.into_bytes()).or_insert(vec![]).push(global_index);
+        self.0.entry(lit.into_bytes()).or_insert_with(|| vec![]).push(global_index);
```

Comment:
I'm not sure I agree with this change either, nor the other uses of `or_insert_with`.